packages/plugin-ext/src/hosted/node/plugin-ext-hosted-backend-module.ts

```diff
@@ -32,44 +32,48 @@ import { GrammarsReader } from './scanners/grammars-reader';
 import { HostedPluginProcess } from './hosted-plugin-process';
 import { ExtPluginApiProvider } from '../../common/plugin-ext-api-contribution';
 import { HostedPluginCliContribution } from './hosted-plugin-cli-contribution';
+import { HostedPluginDeployerHandler } from './hosted-plugin-deployer-handler';
 
-export function bindCommonHostedBackend(bind: interfaces.Bind): void {
-    bind(HostedPluginCliContribution).toSelf().inSingletonScope();
-    bind(CliContribution).toService(HostedPluginCliContribution);
+const commonHostedConnectionModule = ConnectionContainerModule.create(({ bind, bindBackendService }) => {
+    bind(HostedPluginProcess).toSelf().inSingletonScope();
+    bind(HostedPluginSupport).toSelf().inSingletonScope();
 
-    bind(HostedPluginReader).toSelf().inSingletonScope();
+    bind(HostedPluginsManagerImpl).toSelf().inSingletonScope();
+    bind(HostedPluginsManager).toService(HostedPluginsManagerImpl);
 
+    bindContributionProvider(bind, Symbol.for(ExtPluginApiProvider));
     bind(HostedPluginServerImpl).toSelf().inSingletonScope();
     bind(HostedPluginServer).toService(HostedPluginServerImpl);
-    bind(PluginDeployerHandler).toService(HostedPluginServerImpl);
-
-    bind(HostedPluginSupport).toSelf().inSingletonScope();
-    bind(MetadataScanner).toSelf().inSingletonScope();
-    bind(HostedPluginsManager).to(HostedPluginsManagerImpl).inSingletonScope();
+    bindBackendService<HostedPluginServer, HostedPluginClient>(hostedServicePath, HostedPluginServer, (server, client) => {
+        server.setClient(client);
+        return server;
+    });
+});
 
-    bind(HostedPluginProcess).toSelf().inSingletonScope();
+export function bindCommonHostedBackend(bind: interfaces.Bind): void {
+    bind(HostedPluginCliContribution).toSelf().inSingletonScope();
+    bind(CliContribution).toService(HostedPluginCliContribution);
 
+    bind(MetadataScanner).toSelf().inSingletonScope();
+    bind(HostedPluginReader).toSelf().inSingletonScope();
     bind(BackendApplicationContribution).toService(HostedPluginReader);
 
-    bind(ConnectionHandler).toDynamicValue(ctx =>
-        new JsonRpcConnectionHandler<HostedPluginClient>(hostedServicePath, client => {
-            const server = ctx.container.get<HostedPluginServer>(HostedPluginServer);
-            server.setClient(client);
-            // FIXME: handle multiple remote connections
-            /*
-            client.onDidCloseConnection(() => server.dispose());*/
```

Comment:
I guess that server.dispose() would be useful in the new case (as we need to inform somehow that client is not connected if instances are not automatically destroyed ?

Reply:
It should be auto called when connection is closed, i wonder why it does not, will check

Reply:
no, i am wrong, it is not auto disposed, this line is needed, checking again and pushing